### PR TITLE
[bitnami/redis] Fix MASTER invalid argument (using with sentinel)

### LINKED ISSUE #8113: Redis+Sentinel failover failing
Hello,

I'm trying the Helm chart from bitnami/redis. I have 3 nodes redis+sentinel running and i am currently trying to force terminate the redis pod (that is master - in my case master is redis-node-0).
I notice that while this pod is in pending state, the redis cluster stops working (with 2 pods up). As the clusters should only require a quorum of 2 nodes, shoudn't the redis keep working?

Error on one of the slaves: 
```
1:S 09 Nov 2021 16:26:55.496 * Connecting to MASTER redis-node-0.redis-headless.kube-system.svc.cluster.local:6379
1:S 09 Nov 2021 16:26:55.499 # Unable to connect to MASTER: Invalid argument
```
Sentinel:
```
1:X 09 Nov 2021 16:12:30.888 # +switch-master mymaster redis-node-1.redis-headless.kube-system.svc.cluster.local 6379 redis-node-0.redis-headless.kube-system.svc.cluster.local 6379
1:X 09 Nov 2021 16:12:30.889 * +slave slave redis-node-2.redis-headless.kube-system.svc.cluster.local:6379 redis-node-2.redis-headless.kube-system.svc.cluster.local 6379 @ mymaster redis-node-0.redis-headless.kube-system.svc.cluster.local 6379
1:X 09 Nov 2021 16:12:30.889 * +slave slave redis-node-1.redis-headless.kube-system.svc.cluster.local:6379 redis-node-1.redis-headless.kube-system.svc.cluster.local 6379 @ mymaster redis-node-0.redis-headless.kube-system.svc.cluster.local 6379
1:X 09 Nov 2021 16:17:09.498 # +sdown sentinel 2a09ba7abbb41ee71e79087310d75f9809c3c815 redis-node-0.redis-headless.kube-system.svc.cluster.local 26379 @ mymaster redis-node-0.redis-headless.kube-system.svc.cluster.local 6379
1:X 09 Nov 2021 16:17:28.857 # +sdown master mymaster redis-node-0.redis-headless.kube-system.svc.cluster.local 6379
```

Instead the slaves get blocked. Once the master pod comes back the sentinels quickly change the master node to a new pod (but only after the master comes back). What is happening?

 Im using helm with the values bellow. 
 Kubectl server 1.21 client 1.22
```
helm upgrade -i redis bitnami/redis --namespace kube-system --version 15.5.4 --values ~/deletemetoo.yaml
```

```yaml
global:
  storageClass: "default"
  redis:
    password: "PASSWORD"

auth:
  sentinel: false

master:
  resources:
    requests:
      cpu: 100m
    limits:
      cpu: 250m
      memory: 1Gi
  persistence:
    size: 10Gi

replica:
  resources:
    requests:
      cpu: 100m
    limits:
      cpu: 250m
      memory: 1Gi
  persistence:
    size: 10Gi

sentinel:
  enabled: true
  # masterSet: mymaster
  automateClusterRecovery: true
  downAfterMilliseconds: 2000
  failoverTimeout: 1000
  resources:
    requests:
      cpu: 100m
    limits:
      cpu: 200m
      memory: 200Mi    
```

### PR DESCRIPTION
**Pull request by tomislater**
**Description of the change**

Previously we just checked if there are IPs for this service (excluding his own). Here is the code:

```
if [[ -z "$(getent ahosts "$HEADLESS_SERVICE" | grep -v "^${myip}")" ]]; then
```

But sometimes you can got:

```
1:S 03 Jan 2022 12:38:06.207 * Connecting to MASTER :0
1:S 03 Jan 2022 12:38:06.207 # Unable to connect to MASTER: Invalid argument
```

Why? How to reproduce it?

1. Deploy the chart
2. Wait for all pods
3. Kill `node-0` and `node-1`
4. Wait for all pods
5. Now, kill all pods and wait what happen

Because (very likely) `node-2` will be a master and you kill all pods, command `getent ahosts "$HEADLESS_SERVICE" | grep -v "^${myip}"` in `node-0` (after deleting) might return some IPs (because there are still pods in deleting state) and `node-0` will think that it is a replica, and try to get a master address, but in the same time `node-2` might be killed and we will get an empty string from `get_sentinel_master_info`.

```
# Fetches current master's host and port
REDIS_SENTINEL_INFO=($(get_sentinel_master_info))
info "printing REDIS_SENTINEL_INFO=(${REDIS_SENTINEL_INFO[0]},${REDIS_SENTINEL_INFO[1]})"
REDIS_MASTER_HOST=${REDIS_SENTINEL_INFO[0]}
REDIS_MASTER_PORT_NUMBER=${REDIS_SENTINEL_INFO[1]}
```

**Benefits**

You should not see this message anymore:
```
Unable to connect to MASTER: Invalid argument
```

**Possible drawbacks**

I don't know :/

**Applicable issues**

  - fixes #8113

**Additional information**

Values for tests:
```
auth:
  enabled: false
  sentinel: false

master:
  resources:
    requests:
      memory: 128Mi
      cpu: 10m
    limits:
      memory: 128Mi
      cpu: 1
  persistence:
    enabled: false

replica:
  resources:
    requests:
      memory: 128Mi
      cpu: 10m
    limits:
      memory: 128Mi
      cpu: 1
  persistence:
    enabled: false

sentinel:
  enabled: true
  masterSet: b
  automateClusterRecovery: true
  downAfterMilliseconds: 2000
```

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/redis/templates/scripts-configmap.yaml [hunk 1]
```diff
@@ -113,11 +97,11 @@ data:
     [[ -f $REDIS_PASSWORD_FILE ]] && export REDIS_PASSWORD="$(< "${REDIS_PASSWORD_FILE}")"
     [[ -f $REDIS_MASTER_PASSWORD_FILE ]] && export REDIS_MASTER_PASSWORD="$(< "${REDIS_MASTER_PASSWORD_FILE}")"
 
-    # Waits for DNS to add this ip to the service DNS entry
-    retry_while not_exists_dns_entry
```
Comment (miguelaeh) [inline bitnami/redis/templates/scripts-configmap.yaml:117]: Why do we need to remove this check?

Comment (tomislater) [reply bitnami/redis/templates/scripts-configmap.yaml:117]: We don't need to remove this check, but I also don't see any point to check this 🤔

It makes sens to run this if we want to check this command in `if`: `getent ahosts "$HEADLESS_SERVICE" | grep -v "^${myip}"`, but I changed this and now we explicitly get info about master from sentinel.

Comment (miguelaeh) [reply bitnami/redis/templates/scripts-configmap.yaml:117]: 👍

**Review by miguelaeh — APPROVED** [review]
LGTM

**Comment by miguelaeh** [comment]
Please rebase master and solve the conflicts

**Pull request merged**